sdk/src/main/java/io/matchmore/sdk/MatchMore.kt

```diff
@@ -0,0 +1,32 @@
+package io.matchmore.sdk
+
+import android.annotation.SuppressLint
+import io.matchmore.sdk.api.ErrorCallback
+import io.matchmore.sdk.api.SuccessCallback
+import io.matchmore.sdk.api.models.MobileDevice
+
+@SuppressLint("StaticFieldLeak")
+object MatchMore {
+
+    private var matchMoreConfig: MatchMoreConfig? = null
+
+    @JvmStatic
+    val instance: MatchMoreSdk by lazy {
+        if (!isConfigured()) throw IllegalStateException("Please config first.")
+        AlpsManager(matchMoreConfig!!)
+    }
+
+    @JvmStatic
+    fun config(matchMoreConfig: MatchMoreConfig) {
+        if (isConfigured()) throw IllegalStateException("You can not overwrite the configuration.")
+        this.matchMoreConfig = matchMoreConfig
+    }
+
+    @JvmStatic
+    fun isConfigured() = this.matchMoreConfig != null
+}
+
+interface MatchMoreSdk {
+    fun startUsingMainDevice(success: SuccessCallback<MobileDevice>?, error: ErrorCallback?) = startUsingMainDevice(null, success, error)
```

Comment:
is there any smart way to have one callback in kotlin? In swift you can use enum with internal values.

Reply:
As we discussed it's better in kotlin/java to have 2 lambdas.